### DISCUSSION TITLE
Taskbar Icon?

### DISCUSSION THREAD

**Discussion by plettb** [microsoft/WinUI-Gallery#1601] [Q&A] [answered]
I've been playing with the WinUI-Gallery, and learning a LOT!  However, one thing that has me mystified is how the taskbar icon is set.  When I look at the project's properties --> Application --> Win32 Resources --> Icon, it's blank, so it's clearly not there.  Also, properties --> Package --> General --> Icon is blank.  Line 104 in "NavigationRootPage.xaml.cs" is
```
appWindow.SetIcon("Assets/Tiles/GalleryIcon.ico");
```
That definitely looks like it should be the answer.  However, changing it makes no difference.  I've uninstalled the app, cleaned the solution, and rebuilt everything, and the icon persists.

At this point I assume the icon must be cached by Windows, but I have no idea how to clear that cache.

Any help?

**Comment by ghost1372** [comment]
I think you should look for appxManifest file

**Reply by plettb** [reply]
I did - there doesn't appear to be anything in there about the icon.

**Comment by plettb** [comment, marked as answer]
Apparently it's in the Package.appxmanifest, specifically the Square44x44Logo.

**Reply by HEIC-to-JPEG-Dev** [reply]
The package manifest does not alter the Win32 Icon. Currently the app icon is missing.
What the package does is set the titlebar icon (well, one of them, compact view only) - you also need to use SetIcon to get the titlebar working properly.
There is also a bug in registering the Win32 icon when updating msix packages - you have to uninstall/re-install to get teh icon, even when using VS
